### PR TITLE
phaul/StopIter: fixes

### PR DESCRIPTION
**Pull request by kolyshkin**
This fixes three issues in `phaul.StopIter()`:

1. Potential panic in case `s.process` is `nil`.

2. Returning `nil` in case `Wait()` returns an error.

3. `s/s/state/` in the error message.

### CODE REVIEW

**Review by rst0git — APPROVED** [review]

File: phaul/server.go [hunk 1]
```diff
@@ -80,13 +82,16 @@ func (s *Server) StartIter() error {
 
 // StopIter function
 func (s *Server) StopIter() error {
+	if s.process == nil {
+		return errors.New("No process to stop")
+	}
 	state, err := s.process.Wait()
 	if err != nil {
-		return nil
+		return err
```
Comment (rst0git) [inline phaul/server.go:90]: The error could be `ECHILD`, which case the test fails with:
```
Failed: waitid: no child processes
Makefile:39: recipe for target 'phaul-test' failed
```

Comment (kolyshkin) [reply phaul/server.go:91]: Fixed, thanks @rst0git

Comment (rst0git) [reply phaul/server.go:91]:
```diff
diff --git a/phaul/server.go b/phaul/server.go
index 4d1ecf6..b1178b7 100644
--- a/phaul/server.go
+++ b/phaul/server.go
@@ -81,13 +81,27 @@ func (s *Server) StartIter() error {
        return nil
 }
 
+func isNoChildren(err error) bool {
+       switch err := err.(type) {
+       case unix.Errno:
+               if err == unix.ECHILD {
+                       return true
+               }
+       case *os.SyscallError:
+               if err.Err == unix.ECHILD {
+                       return true
+               }
+       }
+       return false
+}
+
 // StopIter function
 func (s *Server) StopIter() error {
        if s.process == nil {
                return errors.New("No process to stop")
        }
        state, err := s.process.Wait()
-       if err != nil && err != unix.ECHILD {
+       if err != nil && !isNoChildren(err) {
                return err
        }
```
based on https://github.com/opencontainers/runc/blob/bb539a9965d3bd0ccb1f46e836cb6c57bb55c3a8/libcontainer/init_linux.go#L458L471

Comment (kolyshkin) [reply phaul/server.go:91]: Ah, I think I know an easier way... please see updated

**Review by rst0git — APPROVED** [review]

**Pull request merged**